core/templates/services/improvements.service.ts

```diff
@@ -30,7 +30,7 @@ export class ImprovementsService {
 
   isStateForcedToResolveOutstandingUnaddressedAnswers(state: State): boolean {
     return !!state && this.INTERACTION_IDS_REQUIRED_TO_BE_RESOLVED.indexOf(
-      state.interaction.id) !== -1;
+      <string>state.interaction.id) !== -1;
```

Comment:
Could you explain why this needs to be cast to a string? Is there any possibility that it might be null?

Reply:
Yep id can be null https://github.com/oppia/oppia/blob/37c24c46e6fc5a9ee1e8df6fb62d55737fa1ced3/assets/constants.ts#L5618
In that case `indexOf` would return -1 and function will return false

Reply:
So why are you casting it to string? That is conceptually wrong, you should instead handle the null case properly.

Reply:
Please also fix this if you have done similar things in this or other PRs -- I'm only catching this because it was in my codeowner file. (Ditto for @EeshaArif.)

Reply:
@seanlip Throwing an error now, PTAL

Reply:
@seanlip we have used typecasting only at places where we we are sure the variable is of that type unless we have missed it unintentionally. Will take a look at the other PRs if this was missed anywhere.

Reply:
Thanks @EeshaArif! So just confirming this was unintentional?

Reply:
@seanlip I probably missed this while taking the pass but have kept this in mind while working on other PRs.